packages/asset-swapper/src/utils/market_operation_utils/types.ts

```diff
@@ -167,4 +164,12 @@ export interface GetMarketOrdersOpts {
      * Default: 1.25.
      */
     sampleDistributionBase: number;
+    /**
+     * Fees for each liquidity source, expressed in ETH wei.
+     * Gas price is not assumed by the optimizer, so these fees should be
+     * pre-scaled by the gas price if one wishes to factor it in.
+     * E.g, if gas costs for a source is 150k and gas price is 1 gwei, the fee
+     * given here should be 150k * 1 gwei.
+     */
+    fees: { [source: string]: BigNumber };
```

Comment:
Hmm on second thought I think we should keep this in terms of gas. And in `swap_quoter.ts` we scale it up before giving it to the `swapQuoteCalculator`. Since at this point we have a `gasPrice` from either the user or from ethgasstation.

Something like: 
```typescript
 let swapQuote: SwapQuote;
        const swapQuoteWithAdjustedFees = {
            ...calculateSwapQuoteOpts,
            fees: Object.assign({}, ...Object.entries(calculateSwapQuoteOpts.fees).map(([k, v]) => ({ [k]: v.times(gasPrice) }))),
        };

        if (marketOperation === MarketOperation.Buy) {
            swapQuote = await this._swapQuoteCalculator.calculateMarketBuySwapQuoteAsync(
                prunedOrders,
                assetFillAmount,
                slippagePercentage,
                gasPrice,
                swapQuoteWithAdjustedFees,
            );
        } else {
            swapQuote = await this._swapQuoteCalculator.calculateMarketSellSwapQuoteAsync(
                prunedOrders,
                assetFillAmount,
                slippagePercentage,
                gasPrice,
                swapQuoteWithAdjustedFees,
            );
        }
```

Reply:
Yeah, I was trying to be agnostic to future fees that aren't necessarily gas price dependent but we don't really need that abstraction right now.